internal/internal_workflow.go

```diff
@@ -696,7 +696,11 @@ func executeDispatcher(ctx Context, dispatcher dispatcher, timeout time.Duration
 			})
 		}
 	}
-	if len(updatesToWarn) > 0 {
+
+	// Verify that the workflow did not fail. If it did we will not warn about unhandled updates.
+	var canceledErr *CanceledError
+	var contErr *ContinueAsNewError
+	if len(updatesToWarn) > 0 && (rp.error == nil || errors.As(rp.error, &canceledErr) || errors.As(rp.error, &contErr)) {
 		env.GetLogger().Warn(unhandledUpdateWarningMessage, "Updates", updatesToWarn)
```

Comment:
You know we never discussed this but perhaps it makes sense to log them at debug level for workflow failing case?

Not a big deal either way.

Reply:
Hmm I think I would punt on this for no. I suspect we will be doing more work on warning on dangling "things" (like activities) in workflow and can resist then